src/sight/core.clj

```diff
@@ -1,67 +1,84 @@
+  [error fileIndex pageNumber numberOfPagesInFile recognizedText])
 (defrecord RecognizedText
-    [text confidence
-     topLeftX topLeftY topRightX topRightY 
-     bottomLeftX bottomLeftY bottomRightX bottomRightY])
+  [text confidence
+   topLeftX topLeftY topRightX topRightY
+   bottomLeftX bottomLeftY bottomRightX bottomRightY])
 
 (defrecord Payload [makeSentences files])
 (defrecord FileEntry [mimeType base64File])
 
+(def ^:private extension->mime-type
+  {:pdf  "application/pdf"
+   :bmp  "image/bmp"
+   :gif  "image/gif"
+   :jpeg "image/jpeg"
+   :jpg  "image/jpg"
+   :png  "image/png"})
+
 (defn slurp-bytes
   "Slurp the bytes from a slurpable thing.
   https://stackoverflow.com/questions/23018870/how-to-read-a-whole-binary-file-nippy-into-byte-array-in-clojure"
   [x]
-  (with-open [out (java.io.ByteArrayOutputStream.)]
+  (with-open [out (ByteArrayOutputStream.)]
     (clojure.java.io/copy (clojure.java.io/input-stream x) out)
     (.toByteArray out)))
 
-(defn filepath->base64file
+(defn file-path->base64file
   "https://stackoverflow.com/questions/42523024/why-is-image-corrupted-when-converted-to-base64
   https://stackoverflow.com/questions/23018870/how-to-read-a-whole-binary-file-nippy-into-byte-array-in-clojure"
-  [filepath]
-  (String. 
-   (Base64/encodeBase64
-    (slurp-bytes filepath))))
+  [file-path]
+  (-> file-path
+      slurp-bytes
+      Base64/encodeBase64
+      String.))
 
-(defn filepath->mimetype
-  [filepath]
-  (if (clojure.string/ends-with? filepath ".pdf") "application/pdf"
-      (if (clojure.string/ends-with? filepath ".bmp") "image/bmp"
-          (if (clojure.string/ends-with? filepath ".gif") "image/gif"
-              (if (clojure.string/ends-with? filepath ".jpeg") "image/jpeg"
-                  (if (clojure.string/ends-with? filepath ".jpg") "image/jpg"
-                      (if (clojure.string/ends-with? filepath ".png") "image/png"
-                          (throw (Exception. "invalid file extension; must be one of \".pdf\", \".bmp\", \".gif\", \".jpeg\", \".jpg\", or \".png\"")))))))))
-
-(defn filepath->file-entry
+(defn- file-extension [file-name]
+  (-> (s/split file-name #"\.")
+      last
+      keyword))
+
+(defn file-path->mimetype
+  [file-path]
+  (if-let [mime-type (-> file-path
+                         file-extension
+                         extension->mime-type)]
+    mime-type
+    (throw (Exception. "invalid file extension; must be one of \".pdf\", \".bmp\", \".gif\", \".jpeg\", \".jpg\", or \".png\""))))
+
+(defn file-path->file-entry
   [filepath]
-  (->FileEntry (filepath->mimetype filepath) (filepath->base64file filepath)))
+  (->FileEntry (file-path->mimetype filepath)
+               (file-path->base64file filepath)))
 
-(defn filepaths->file-entries
-  [filepaths]
-  (if (empty? filepaths) '()
-      (apply list (filepath->file-entry (first filepaths)) (filepaths->file-entries (rest filepaths)))))
+(defn file-paths->file-entries
+  [& file-paths]
```

Comment:
To make a long story short, let's remove the `&` here.

### Long Story

On line 75, `file-paths` is passed into this function as a list:

```
(file-paths->file-entries file-paths)))
```

which means the `&` in `[& file-paths]` makes `file-paths` a list of a list. Like this:

```
(list (list "image.jpeg"))
```

which means `(map file-path->file-entry file-paths)` fails on line 70.

This is all avoided, by passing in strings instead of a list of strings to `recognize`. For example,

```
(recognize client "image.jpeg" "image2.png")
```
instead of the current interface:

```
(recognize client (list "image.jpeg" "image2.png"))
```

While the first way of doing things may look cleaner, I prefer not to change the signature of the API from version to version. I applaud your choice and I think it is cleaner code, but I ultimately would prefer uglier code to changing the interface, in case anyone uses this already.